sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+
+class ConfigurationOutput:
+    """Class that collects configuration data and outputs it a json format."""
+    solver: Solver = None
+    configurator: Configurator = None
+    best_config: str = None
+    training: ConfigurationResults = None
+    test: ConfigurationResults = None
+
+    instance_set_test: InstanceSet = None
+
+    def __init__(self: ConfigurationOutput, path: Path, solver_name: str,
+                 configurator: Configurator, penalty_multiplier: int,
+                 output: Path = None) -> None:
+        """Initialize Configurator Output class.
+
+        Args:
+            path: Path to configuration directory
+            solver_name: Name of the solver that was used
+            configurator: The configurator that was used
+            penalty_multiplier: penalty multiplier that is applied to the par performance
+            output: Path to the output directory
+        """
+        self.solver = resolve_object_name(solver_name,
+                                          gv.solver_nickname_mapping,
+                                          gv.solver_dir, Solver)
+        self.configurator = configurator
+        self.penalty_multiplier = penalty_multiplier
+        self.directory = path
+
+        if output is None:
+            output = path / "Analysis" / "configuration.json"
+        self.output = output
+
+        # TODO: Fix this spaghetti code to find the scenario file
+        solver_dir_name = path.name
+        scenario_dir = path / "outdir_train_configuration" / \
+            f"{solver_dir_name}_scenario" / "state-run0"
+        if scenario_dir.is_dir():
+            scenario_file = scenario_dir / "scenario.txt"
+            if not scenario_file.is_file():
+                print("Can't find scenario file")
+                return
+        else:
+            print("Can't find scenario file")
+            return
+
+        # Sets scenario on configurator object
+        self.initalise_scenario(scenario_file)
+        # Retrieves validation results and best configuration
+        self.training, self.best_config =\
+            self.get_validation_data(self.configurator.scenario.instance_set)
+        # Retrieve test validation results if they exist
+        if self.instance_set_test is not None:
+            self.test, _ = self.get_validation_data(self.instance_set_test)
+
+        self.write_output()
+
+    def initalise_scenario(self: ConfigurationOutput, scenario_file: Path) -> None:
+        """Reads scenario file and initalises ConfigurationScenario."""
+        config = {}
+        with scenario_file.open() as file:
+            for line in file:
+                key, value = line.strip().split(" = ")
+                config[key] = value
+
+        instance_set_train = resolve_object_name(
+            config["instance_file"],
+            gv.file_storage_data_mapping[gv.instances_nickname_path],
+            gv.instance_dir, InstanceSet)
+        # TODO: Find out why _train.txt is added to end of name
```

Comment:
Good question, you can take a look at ConfigurationScenario. Could be legacy code, and if so we can remove it.